### PR TITLE
[feature][expression] Add exif() and exif_geotag() functions to the QGIS expression engine

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR adds a pair of functions to the QGIS expression engine to retrieve exif tags from an image file. The two functions are:
- exif(path, tag) : this function returns the value of a given tag string for the provided image file path; when the second optional parameter isn't  provided, the function will return a map object containing _all_ exif tags and their values.
- exif_geotag(path) : this function returns a point geometry from the exif geotags containing in the provided image file path.

### CODE REVIEW

File: src/core/raster/qgsexiftools.cpp [hunk 1]
```diff
@@ -171,42 +171,62 @@ QString doubleToExifCoordinateString( const double val )
...
+      const QString key = QString::fromStdString( i->key() );
+      res.insert( key, decodeExifData( key, i ) );
+    }
+    return res;
+  }
+  catch ( ... )
   {
-    const QString key = QString::fromStdString( i->key() );
-    res.insert( key, decodeExifData( key, i ) );
+    return QVariantMap();
```
Comment (m-kuhn) [inline src/core/raster/qgsexiftools.cpp:228]: Is it feasible to report an error here?

**Review by m-kuhn — APPROVED** [review]
I like

File: src/core/raster/qgsexiftools.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/raster/qgsexiftools.cpp:228]: Exiv2 throws quite a few exceptions, including when a key string isn't found in the image (which I think we should be tolerant about). I personally see little use in error reporting here. Maybe we defer until someone has an actual use for some error reporting?

**Comment by nirvn** [comment]
One very nice use of this new feature is the auto-filling of feature attributes when a photo is added via an external resource widget.

File: src/core/raster/qgsexiftools.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/raster/qgsexiftools.cpp:228]: sounds good

**Pull request merged**